library/system/src/lib/yast2/system_service.rb

```diff
@@ -0,0 +1,301 @@
+# encoding: utf-8
+
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "forwardable"
+Yast.import "SystemdService"
+
+module Yast2
+  # This class represents a system service
+  #
+  # When talking about systemd, it might happen that a service is compose by a set of units
+  # (services, sockets, paths and so on). This class is able to group those units and
+  # offer an API to handle them together.
+  #
+  # Additionally, this class does not modify the underlying system until the #save
+  # method is called.
+  #
+  # @example Enabling a service
+  #   cups = SystemService.find("cups")
+  #   cups.start_mode = :boot
+  #   cups.save
+  #
+  # @example Activating a service
+  #   cups = SystemService.find("cups")
+  #   cups.start
+  #   cups.save
+  #
+  # @example Ignoring status changes on 1st stage
+  #   cups = SystemService.find("cups")
+  #   cups.start
+  #   cups.save(ignore_status: true)
+  class SystemService
+    extend Forwardable
+
+    # @return [Yast::SystemdService]
+    attr_reader :service
+
+    def_delegators :@service, :running?, :name, :description
+
+    # @!method state
+    #
+    # @return [String]
+    def_delegator :@service, :active_state, :state
+
+    # @!method substate
+    #
+    # @return [String]
+    def_delegator :@service, :sub_state, :substate
+
+    class << self
+      # Find a service
+      #
+      # @param name [String] Service name
+      # @return [SystemService,nil] System service or nil when not found
+      def find(name)
+        new(Yast::SystemdService.find(name))
+      end
+
+      # Finds service names
+      #
+      # This method finds a set of system services. Currently it is just a wrapper around
+      # SystemdService.find_many.
+      #
+      # @param names [Array<String>] Service names to find
+      # @return [Array<SystemService>] Found system services
+      def find_many(names)
+        Yast::SystemdService.find_many(names).compact.map { |s| new(s) }
```

Comment:
to be honest, I do not like much here that it just ignore names that is not found. why not return in map nils?